overlord/ifacestate/helpers.go

```diff
@@ -236,12 +236,37 @@ func (m *InterfaceManager) autoConnect(task *state.Task, snapName string, blackl
 			continue
 		}
 		if err := m.repo.Connect(connRef); err != nil {
-			task.Logf("cannot auto connect %s to %s: %s", connRef.PlugRef, connRef.SlotRef, err)
+			task.Logf("cannot auto connect %s to %s: %s (plug auto-connection)", connRef.PlugRef, connRef.SlotRef, err)
+			continue
 		}
 		affectedSnapNames = append(affectedSnapNames, connRef.PlugRef.Snap)
 		affectedSnapNames = append(affectedSnapNames, connRef.SlotRef.Snap)
 		conns[key] = connState{Interface: plug.Interface, Auto: true}
 	}
+	// Auto-connect all the slots
+	for _, slot := range m.repo.Slots(snapName) {
```

Comment:
This loops looks very similar to the loop above, is there a way to extra common parts maybe? Not sure if its worth it given that they are placed right next to each other but there is always the risk that they diverge in the future because things get only (accidentally) fixed in a single place.